### ISSUE TITLE
Bug: xlog flush request $LSN is not satisfied 

### ISSUE THREAD

**Issue by kelvich** [zenithdb/zenith#225]
Few times our tests caught situation where regression tests failed with "xlog flush request 0/EF8C018 is not satisfied". It seems that problem is caused by some erroneous flush logic in zenith_get_request_lsn().

https://app.circleci.com/pipelines/github/zenithdb/zenith/594/workflows/13555ca1-af34-4cab-927e-71253bc9e702/jobs/2554/tests

**Comment by knizhnik** [comment]
Suggested fix in  adjust_last_written_lsn branch.
I am not sure that it is real source of the problem and this fix is enough.
It is only my guess. Since the bug is very bad reproduced, it is not so easy to prove it with test.

**Comment by kelvich** [comment]
https://github.com/zenithdb/postgres/commit/d6dd20a78bbdfa0680c9c6ef5e7bc59a36d05a16 @hlinnaka could you take a look?

**Comment by knizhnik** [comment]
We are using GetXLogInsertRecPtrto remember last written lsn. It uses  XLogBytePosToRecPtr which advance pointer beyond page header:
```
		seg_offset += fullpages * XLOG_BLCKSZ + bytesleft + SizeOfXLogShortPHD;
```
So looks like we either need to use some other function , either return pointer backward to the beginning of the page.

**Comment by lubennikovaav** [comment]
I took a look at the fix in zenithdb/postgres@d6dd20a and I think the fix itself is correct but it should be placed in another function, which is the source of the problem.
See branch adjust_last_written_lsn_in_zenith_truncate

I'll write a test that generates such a record to confirm that you found reason of the problem correctly.
